tests/lib/thrift/ThriftTest/CMakeLists.txt

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 Meta
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(thrift_test)
+
+set(THRIFT_UPSTREAM ${ZEPHYR_THRIFT_MODULE_DIR})
+
+include(${ZEPHYR_BASE}/modules/thrift/cmake/thrift.cmake)
+
+FILE(GLOB app_sources
+    src/*.cpp
+)
+
+set(generated_sources "")
+set(gen_dir ${ZEPHYR_BINARY_DIR}/misc/generated/thrift_ThriftTest)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/SecondService.cpp)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/SecondService.h)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/ThriftTest_constants.cpp)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/ThriftTest_constants.h)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/ThriftTest.cpp)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/ThriftTest.h)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/ThriftTest_types.cpp)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/ThriftTest_types.h)
+list(APPEND app_sources ${generated_sources})
+
+thrift(
+    app
+    cpp
+    :no_skeleton
+    ${gen_dir}
+    # v0.16: ubuntu packaged thrift compiler does not support 'uuid' type
+    "${THRIFT_UPSTREAM}/test/v0.16/ThriftTest.thrift"
```

Comment:
I assume that this isn't really thrift upstream and is actually just the normal zephyr version of the thrift repository that requires manual manifest updates? If this really is the upstream, seems like hard coding the version here is going to break

Reply:
It won't break anything, as the upstream commit is pinned in the west manifest 😄.

As with everything, we have created a fork. Yes, it contains a Zephyr manifest. The fork is entirely redundant and was only made because that is Zephyr's policy.

In any case though, there was a slight .thrift file syntax change after v0.16 where a "native" uuid type was added. However, since all of the existing deployed packages still used the v0.16 syntax version pre-uuid support, they were seeing a lot of problems downstream. Their solution was to keep a version of tests pinned during the transition and then later deprecated.